origins_workflow/origins_workflow.module

```diff
@@ -278,3 +266,106 @@ function _audit_link($field, $dt, $nid) {
     }
   }
 }
+
+/**
+ * Implements hook_preprocess_table().
+ */
+function origins_workflow_preprocess_table(&$variables) {
+  // Add new 'Revision Number' column to the node revisions tab.
+  //
+  // Making this change in preprocess is not beautiful, but seems
+  // to be the only workable solution as the 'diff' module has
+  // already been enabled and has taken over the core route with
+  // its own controller. If we created our own controller to do
+  // this then we would have to call the contrib controller,
+  // which does not sound like a good idea.
+  if (\Drupal::routeMatch()->getRouteName() == 'entity.node.version_history') {
+    // Add 'Revision Number' column header.
+    $new_head = ['tag' => 'th', 'content' => ['#markup' => t('Revision number')]];
+    // Insert this new column after the first one.
+    // Use array_splice as we don't know exactly how many columns there will be
+    // (and this code should work whether the 'diff' module has been enabled or
+    // not).
+    array_splice($variables['header'], 1, 0, [$new_head]);
+    if (isset($variables['header']['select_column_one'])) {
+      $variables['header']['select_column_one']['content'] = ['#markup' => t('Compare revision 1')];
+    }
+    if (isset($variables['header']['select_column_two'])) {
+      $variables['header']['select_column_two']['content'] = ['#markup' => t('Compare revision 2')];
+    }
+    // Loop through the rows and add the revision number column to each.
+    foreach ($variables['rows'] as $idx => $row) {
+      $this_version = '';
+      if (isset($row['cells'][3]['content']['#links'])) {
+        // Cheekily extract current revision from the 'revert' button link.
+        $this_url = $row['cells'][3]['content']['#links']['revert']['url'];
+        if (isset($this_url)) {
+          $this_version = $this_url->getRouteParameters()['node_revision'];
+        }
+        // Change the link text from 'Set as current revision' to
+        // 'Copy to new revision' as it makes more sense - see issue
+        // https://www.drupal.org/project/drupal/issues/2899719
+        if (isset($variables['rows'][$idx]['cells'][3]['content']['#links']['revert']['title'])) {
+          $title_markup = $variables['rows'][$idx]['cells'][3]['content']['#links']['revert']['title'];
+          // Check the existing text in an attempt to detect any core fix that
+          // might come along.
+          $title_text = $title_markup->getUntranslatedString();
+          if (($title_text == 'Set as current revision') || ($title_text == 'Revert')) {
+            $variables['rows'][$idx]['cells'][3]['content']['#links']['revert']['title'] = t('Copy to new revision');
+          }
```

Comment:
This is new code to change the link text to the more logical 'Copy to new revision'